common/inc/ConnectionThingy.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Mbed-OS Microcontroller Library
+ * Copyright (c) 2020 Embedded Planet
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+#ifndef MBED_OS_EXPERIMENTAL_BLE_SERVICES_COMMON_INC_CONNECTIONTHINGY_H_
+#define MBED_OS_EXPERIMENTAL_BLE_SERVICES_COMMON_INC_CONNECTIONTHINGY_H_
+
+/**
+ * Connection... Thingy?
+ * A common problem encountered when developing BLE applications is:
+ *
+ * Characteristic read/write/notify/indicate events are handled in batch.
+ *
+ * This creates a lot of boilerplate code where a service must filter out by comparing
+ * which handle was written to the callback's parameters, figure out which connection,
+ * and then perform the appropriate action.
```

Comment:
Do [`setWriteAuthorizationCallback`](https://github.com/ARMmbed/mbed-os/blob/aeaac0e70c76fbc5b753ee8c836e369edf1a7fb4/connectivity/FEATURE_BLE/include/ble/gatt/GattCharacteristic.h#L1556) and [`setReadAuthorizationCallback`](https://github.com/ARMmbed/mbed-os/blob/aeaac0e70c76fbc5b753ee8c836e369edf1a7fb4/connectivity/FEATURE_BLE/include/ble/gatt/GattCharacteristic.h#L1603) not solve this problem?

Reply:
Unrelated the PR at hand. Those APIs "solve" the problem, but I think it creates a confusing situation in the BLE API:
You can still handle characteristic writes that are authorized but you then have to filter them based on the value handle. It creates two ways to do the same thing which can be confusing to users.

The naming of "setWriteAuthorizationCallback" and "setReadAuthorizationCallback" imply they are _only_ for read authorization/write authorization. Likewise, the naming of `GattServer::EventHandler::onCharacteristicWritten` (or whatever it is) implies it should be where characteristic writes are handled.